### PR TITLE
Make Menu.get_template neutral to template engine

### PR DESCRIPTION
**Pull request by hongquan**
This is to make _wagtailmenus_ work with Jinja2 template engine (rkhleics/wagtailmenus#89).

The PR makes change in one internal API: The `Menu.get_template()` now return instance of `Template` class in `django.template.backends`, not `django.template.base.Template` class.

### CODE REVIEW

**Comment by ababic** [comment]
Hi @hongquan,

Thank you for this. It looks as though the Travis CI integration hasn't worked for this PR (I'm unsure why). Would you possibly be able to make a very minimal commit, and we'll see if Travis CI will do it's thing?

Thanks,
Andy

**Comment by hongquan** [comment]
I've just added 1 commit.

**Comment by ababic** [comment]
Hi @hongquan,

Travis CI is failing to build with 'Abuse detected' for some reason. I've just opened a ticket with their support team to see why this is happening.

**Comment by ababic** [comment]
Hi @hongquan,

I have the following feedback from the Travis team:

> If a user- or the organization the pull request is coming from- hasn't ever used Travis before, our system will automatically flag the pull request. It looks like no user with push access to the AgriConnect org has used Travis- at least, not since that org was created. The solution here would be for the user who created the pull request to log into travis-ci.org and sync their account- after that, they should be good go, and if that doesn't work, they'll be in our system so we can manually flag it as not abusive.

Are you okay to try signing up for an account on travis-ci.org using you github account? It sounds like that might solve the problem.

Thanks :)

**Comment by hongquan** [comment]
I have just grant Travis CI access to AgriConnect. Hope it will help.
screenshot from 2017-11-12 19-10-16

**Comment by ababic** [comment]
Thanks @hongquan. If you could just make another small commit, we'll see if it worked :) maybe just change your name in contributors, then commit a fix for it after?

**Comment by hongquan** [comment]
I have to do this in Travis CI to make it able to build:

screenshot from 2017-11-13 11-01-51

**Comment by ababic** [comment]
Hi @hongquan,

Thank you for your time in putting this together. I'm happy enough with the code changes here, but I'm not quite sure about making these additions to the documentation just yet - I think I would rather those be added separately as another pull, along with some other things, such as:

1. Adding a complete Jinja extension to wagtailmenus (making all tags available, not just `flat_menu` and `sub_menu`) so that users don't need to create their own.
2. Updating the test suite settings to support Jinja templates and make use of the extension.
3. Adding Jinja page/menu templates to the test app and testing the HTML output of all tags to ensure the output is as expected. 

The instructions can then be simplified, and we'll have a working settings configuration that we can point developers to in case they'd rather look at / copy from a working implementation.

Does this sound okay to you?

**Comment by hongquan** [comment]
It is OK to me. Do I need to revert the commit of updating documentation?

**Comment by ababic** [comment]
@hongquan Yes, if you wouldn't mind :)

File: docs/source/rendering_menus/index.rst [hunk 1]
```diff
@@ -7,3 +7,4 @@ Rendering menus
 
     template_tag_reference
     custom_templates
+    use_with_jinja2
```
Comment (ababic) [inline docs/source/rendering_menus/index.rst]: Please remove this for now - We'll work on Jinja-specific stuff as a separate piece of work :)

File: docs/source/rendering_menus/use_with_jinja2.rst [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+4. In your template content, insert menu like this:
+
+.. code-block:: jinja
+
+    <div class="collapse navbar-collapse" id="myNavbar">
+      {{ main_menu() }}
+    </div>
+
+
```
Comment (ababic) [inline docs/source/rendering_menus/use_with_jinja2.rst]: Please remove this file for now - We'll work on Jinja-specific stuff as a separate piece of work :)

**Comment by hongquan** [comment]
I removed the docs.

**Comment by ababic** [comment]
Big thanks @hongquan :) This has now been merged

**Pull request merged**